### PR TITLE
KAFKA-17765: Remove TestUtils.toSet

### PR DESCRIPTION
**Pull request by Linsiyuan9**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ public void deserializeOldSubscriptionVersion() {
         Subscription subscription = new Subscription(Arrays.asList("foo", "bar"), null);
         ByteBuffer buffer = ConsumerProtocol.serializeSubscription(subscription, (short) 0);
         Subscription parsedSubscription = ConsumerProtocol.deserializeSubscription(buffer);
-        assertEquals(toSet(parsedSubscription.topics()), toSet(parsedSubscription.topics()));
+        assertEquals(Set.copyOf(parsedSubscription.topics()), Set.copyOf(parsedSubscription.topics()));
```
Comment (Linsiyuan9) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java]: This should be a wrong unit tests, expect code should be `assertEquals(Set.copyOf(subscription.topics()), Set.copyOf(parsedSubscription.topics()));`

**Comment by chia7712** [comment]
@Linsiyuan9 please fix the title

**Comment by Linsiyuan9** [comment]
Hi @chia7712,Have been modified.

**Review by frankvicky — APPROVED** [review]
LGTM

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java (hunk 1 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java]: yes, you are right. Could you please fix it in this PR?

Comment (Linsiyuan9) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java]: Of course, I have pushed the repair code, please have a look

**Comment by chia7712** [comment]
the failed test is traced by https://issues.apache.org/jira/browse/KAFKA-17765

**Review by chia7712 — APPROVED** [review]

**Pull request merged**